src/tree-view/treeViewComponent.ts

```diff
@@ -44,8 +44,8 @@ export class TreeViewController {
         showBorders:     false,
         onNodeExpanded:  this.storeNodeState(true),
         onNodeCollapsed: this.storeNodeState(undefined),
-        onNodeSelected:  (_event, node) => this.$timeout(() => this.onSelect({node: node})),
-        lazyLoad:        (node, render) => this.$timeout(() => this.lazyLoad({node: node}).then(render)),
+        onNodeSelected:  this.onSelect ? (_event, node) => this.$timeout(() => this.onSelect({node: node})) : () => null,
```

Comment:
These lines are too long for the linter :cry:

Reply:
@karelhala can't those be set as default values at the type declaration?

Reply:
Either better space align or, we can use private methods for these 2 functions
```typescript
  private onSelected(node) {
    this.$timeout(() => this.onSelect({node: node}));
  }

  private onLazyLoad(node, render) {
    this.$timeout(() => this.lazyLoad({node: node}).then(render));
  }

  private renderTree() {
    new Promise((resolve) => {
      this.element.treeview({
        //...
        onNodeSelected:  this.onSelect ? (_event, node) => this.onSelected(node) : () => null,
        lazyLoad:        this.lazyLoad ? (node, render) => this.onLazyLoad(node, render) : () => null,
        //...
      }
    }
  }
```

@himdel what do you think?

Reply:
Well, I think `$timeout` needs to live in that `treeview` call and not somewhere else.. and after that, there's not much point in creating functions.

So.. I'd just disable the long lines linter rule :).

But.. what aobut just providing a default value for `onSelect` and `lazyLoad`? That way, no conditionals necesarry...

Reply:
Default values :+1:, you have to put them into the constructor as it doesn't work at the declaration (tried, failed)